numaprom/clients/redis.py

```diff
@@ -25,11 +27,13 @@ def get_redis_client(
         "decode_responses": decode_responses,
         "skip_full_coverage_check": True,
         "dynamic_startup_nodes": False,
+        "cluster_error_retry_attempts": 3,
     }
     _LOGGER.info("Redis params: %s", json.dumps(redis_params, indent=4))
 
     if not is_host_reachable(host, port):
         _LOGGER.error("Redis Cluster is unreachable after retries!")
 
-    redis_client = RedisCluster(**redis_params)
+    retry = Retry(ExponentialBackoff(cap=2, base=1), 3)
+    redis_client = RedisCluster(**redis_params, retry=retry)
```

Comment:
does this retry happen only on cluster errors or the other errors too?

Reply:
Good catch, adding errors

Reply:
with this do we still need to catch exceptions while calling `get_redis_client`?

Reply:
Unfortunately, yes. Since this is only while creating a connection. As discussed, we need to wrap around the redis function calls to catch the exception ourselves :/

Reply:
aah okay, I think instead on recreate boolean, we should may be use the `redis_client.ping() `to check if its available, like the stackoverflow solution you shared